packages/twenty-server/src/workspace/workspace-schema-builder/utils/get-resolver-args.util.ts

```diff
@@ -69,6 +69,17 @@ export const getResolverArgs = (
           isNullable: false,
         },
       };
+    case 'findDuplicates':
+      return {
+        id: {
+          type: FieldMetadataType.UUID,
+          isNullable: true,
+        },
+        data: {
+          kind: InputTypeDefinitionKind.Update,
```

Comment:
This type is not ideal. I dont want to use `InputTypeDefinitionKind.Filter` though, because we wont allow specifying filter types. Am I right? What would be your suggestion to use here? 🤔

Reply:
I totally agree with you on that.
Also this `data` input should be limited to the provided `criteriaCollection` not all the fields of the object.
Only providing the fields exposed by `criteriaCollection` is adding more complexity because we need to create a new graphQL input type based on the `criteriaCollection`, and this one is using `columnNames` and not `fieldNames`.
Do we keep it that way for now @charlesBochet ?

Reply:
"Also this data input should be limited to the provided criteriaCollection not all the fields of the object."
Not sure I understand/agree with that? The de-duplication logic is server-side, it's not the frontend's role to stay up-to-date on which field or not might trigger a duplicate alert (it should be allowed to send everything they want on that object), that logic will evolve and be different for each workspace long-term.

Reply:
I agree with `columnNames` rather than `fieldNames`!

Reply:
@rostaklein Ok we'll keep it that way for now :)